### PR TITLE
Add catalogAddMenu filter to CatalogCategory

### PR DESCRIPTION
**Pull request by panuhorsmalahti**
* Add filter for `catalogAddMenu` context menu items. Extension can e.g. filter certain menu items from the context menu.
* Example use:
```
const disposer = kubernetesClusterCategory.addMenuFilter(item => item.title === "Add from kubeconfig");
```
The filter can be removed with the returned disposer:
```
disposer();
```
* This is supported for all `CatalogCategory` classes.

### CODE REVIEW

**Review by chenhunghan — DISMISSED** [review]
LGTM

File: src/common/catalog-entities/__tests__/kubernetes-cluster.test.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import { kubernetesClusterCategory }  from "../kubernetes-cluster";
+
+describe("kubernetesClusterCategory", () => {
+  describe("filteredItems", () => {
```
Comment (chenhunghan) [inline src/common/catalog-entities/__tests__/kubernetes-cluster.test.ts:24]: maybe it's more useful to add test for `<CatalogAddButton  />`, but I guess this is fine too..

Comment (panuhorsmalahti) [reply src/common/catalog-entities/__tests__/kubernetes-cluster.test.ts:24]: True, tests added

**Review by chenhunghan — DISMISSED** [review]
Looks great

File: src/common/catalog/catalog-entity.ts [hunk 2]
```diff
@@ -72,6 +83,32 @@ export abstract class CatalogCategory extends (EventEmitter as new () => TypedEm
...
+
+  /**
+   * Add a filter for menu items of catalogAddMenu
+   * @param fn The function that should return a truthy value if that menu item should be displayed
+   * @returns A function to remove that filter
+   */
+  public addMenuFilter(fn: AddMenuFilter): Disposer {
+    this.filters.add(fn);
+
+    return once(() => void this.filters.delete(fn));
```
Comment (Nokel81) [inline src/common/catalog/catalog-entity.ts:95]: Only nit would be that if the category is removed, these should be removed too.

**Review by Nokel81 — APPROVED** [review]

File: src/common/catalog/catalog-entity.ts (hunk 2 above)
Comment (panuhorsmalahti) [reply src/common/catalog/catalog-entity.ts:95]: Why? Do you mean if I call the `Disposer` returned by `CatalogCategoryRegistry::add`, I should remove the filters from the `CatalogCategory` instance? Doesn't sound right to me

Comment (Nokel81) [reply src/common/catalog/catalog-entity.ts:95]: Yeah probably not.

**Pull request merged**